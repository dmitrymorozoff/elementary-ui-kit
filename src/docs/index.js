import React from "react";
import ReactDOM from "react-dom";
import {
    Modal,
    ModalContent,
    ModalTitle,
    Button,
    ModalActions,
    ModalMedia,
} from "../lib";

class App extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            open: false,
        };
    }

    handleClickOpen() {
        this.setState({ open: true });
    }

    handleClose() {
        this.setState({ open: false });
    }

    render() {
        const { open } = this.state;
        return (
            <React.Fragment>
                <Button onClick={this.handleClickOpen.bind(this)}>
                    Open Modal
                </Button>
                <Modal
                    className="my-modal"
                    closeOnEsc={true}
                    open={open}
                    onClose={this.handleClose.bind(this)}
                >
                    <ModalMedia image="https://images.unsplash.com/photo-1486546910464-ec8e45c4a137?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=a2c308fe62dad65bdfd4b1efb8bc6bb8&auto=format&fit=crop&w=2502&q=80" />
                    <ModalTitle>Hey now, you're an allstar</ModalTitle>
                    <ModalContent>
                        Somebody once told me the world is gonna roll me. I
                        ain't the sharpest tool in the shed. She was looking
                        kind of dumb with her finger and her thumb in the shape
                        of an "L" on her forehead.
                        <br />
                        <br />
                        Well the years start coming and they don't stop coming.
                        Fed to the rules and I hit the ground running.
                    </ModalContent>
                    <ModalActions>
                        <Button margin="5px">Disagree</Button>
                        <Button margin="5px" type="primary">
                            Agree
                        </Button>
                    </ModalActions>
                </Modal>
            </React.Fragment>
        );
    }
}

// eslint-disable-next-line
ReactDOM.render(<App />, document.getElementById("root"));
