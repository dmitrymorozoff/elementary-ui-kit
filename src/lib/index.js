import { Button } from "./components/forms/Button";
import { Link } from "./components/forms/Link";
import { Switch } from "./components/forms/Switch";
import { Input } from "./components/forms/Input";
import { Checkbox } from "./components/forms/Checkbox";
import { Card } from "./components/others/Card";
import { Label } from "./components/others/Label";
import { CardButtons } from "./components/others/Card/components/CardButtons";
import { CardContent } from "./components/others/Card/components/CardContent";
import { CardMedia } from "./components/others/Card/components/CardMedia";
import { Collapse } from "./components/others/Collapse";
import { CollapseContent } from "./components/others/Collapse/components/CollapseContent";
import { CollapseTitle } from "./components/others/Collapse/components/CollapseTitle";
import { Modal } from "./components/others/Modal";
import { ModalContent } from "./components/others/Modal/components/ModalContent";
import { ModalTitle } from "./components/others/Modal/components/ModalTitle";
import { ModalActions } from "./components/others/Modal/components/ModalActions";
import { Text } from "./components/others/Text";
import { Search } from "./components/icons/Search";
import { RightArrow } from "./components/icons/RightArrow";
import { Plus } from "./components/icons/Plus";
import { DownArrow } from "./components/icons/DownArrow";
import { UpArrow } from "./components/icons/UpArrow";
import { Checked } from "./components/icons/Checked";
import { Select } from "./components/forms/Select";
import { SelectItem } from "./components/forms/Select/components/SelectItem";

export {
    Button,
    Link,
    Switch,
    Input,
    Checkbox,
    Card,
    CardButtons,
    CardContent,
    CardMedia,
    Text,
    Search,
    RightArrow,
    DownArrow,
    UpArrow,
    Checked,
    Select,
    SelectItem,
    Label,
    Plus,
    Collapse,
    CollapseContent,
    CollapseTitle,
    Modal,
    ModalContent,
    ModalTitle,
    ModalActions,
};
