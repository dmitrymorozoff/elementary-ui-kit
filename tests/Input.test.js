import React from "react";
import { Input } from "../src/lib";

describe("<Input/>", () => {
    it("should render a default input", () => {
        const component = shallow(<Input value="" onChange={() => {}} />);
        expect(component).toMatchSnapshot();
    });
    it("it input change correctly", () => {
        const value = "Hello";
        const mockFunction = jest.fn();
        const component = mount(<Input value="" onChange={mockFunction} />);
        component.find("input").simulate("change", {
            preventDefault() {},
            target: { value },
        });
        expect(mockFunction).toHaveBeenCalled();
    });
});
