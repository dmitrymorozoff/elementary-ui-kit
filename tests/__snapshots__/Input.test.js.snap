// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input/> should render a default input 1`] = `
<styled.div
  color="#3A77F8"
  disabled={false}
  error={false}
  fullWidth={false}
  hasFocus={false}
  margin="0px"
  onChange={[Function]}
  required={false}
  type="text"
  value=""
>
  <styled.input
    color="#3A77F8"
    disabled={false}
    error={false}
    fullWidth={false}
    leftIconComponent={null}
    margin="0px"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    required={false}
    rightIconComponent={null}
    type="text"
    value=""
  />
</styled.div>
`;
